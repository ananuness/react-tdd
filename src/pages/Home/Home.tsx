import { DrawButton } from '../../components/DrawButton/DrawButton';
import { Form } from '../../components/Form/Form';
import { ParticipantsList } from '../../components/ParticipantsList/ParticipantsList';

export const Home = () => {
  return (
    <>
      <Form />
      <ParticipantsList />
      <DrawButton />
    </>
  );
};