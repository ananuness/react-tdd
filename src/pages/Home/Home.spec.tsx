import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import { Home } from './Home';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigate
  };
});

describe('Snapshot of Home page', () => {
  it('should be rendered correctly', () => {
    const { container } = render(
      <RecoilRoot>
        <Home />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });

  afterAll(() => {
    jest.resetAllMocks();
  });
});