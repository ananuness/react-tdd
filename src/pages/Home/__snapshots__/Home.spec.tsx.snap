// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Home page should be rendered correctly 1`] = `
<div>
  <form>
    <input
      placeholder="New participant"
      type="text"
      value=""
    />
    <button
      disabled=""
    />
  </form>
  <ul />
  <button
    class="draw-button"
    disabled=""
  >
    Sortear!
  </button>
</div>
`;
